src/jit/codegenarmarch.cpp

```diff
@@ -1617,11 +1766,29 @@ void CodeGen::genCallInstruction(GenTreeCall* call)
         }
         else
         {
-            regNumber argReg = curArgTabEntry->regNum;
             genConsumeReg(argNode);
```

Comment:
If it is the case that a `fgArgTabEntry` with `isSplit == true` will always have a `GT_PUTARG_SPLIT`, then you should assert it here. In that case, please add a comment to indicate that a `genConsumeReg` is not needed, and move the `genConsumeReg` above to the else clause below here. Otherwise it makes no sense that you have consumed just a single register for a node that has multiple registers.